src/Stratis.Bitcoin.Features.SmartContracts/SmartContractFeature.cs

```diff
@@ -141,19 +141,86 @@ public static IFullNodeBuilder UseSmartContractConsensus(this IFullNodeBuilder f
                     services.AddSingleton<ConsensusStats>();
                     services.AddSingleton<ConsensusSettings>();
 
-                    services.AddSingleton<IConsensusRules, SmartContractConsensusRules>();
-                    services.AddSingleton<IRuleRegistration, SmartContractRuleRegistration>();
+                    services.AddSingleton<IConsensusRules, SmartContractPowConsensusRuleEngine>();
+                    services.AddSingleton<IRuleRegistration, SmartContractPowRuleRegistration>();
                 });
             });
 
             return fullNodeBuilder;
         }
 
+        /// <summary>
+        /// Configures the node with the smart contract proof of stake consensus model.
+        /// </summary>
+        public static IFullNodeBuilder UseSmartContractPosConsensus(this IFullNodeBuilder fullNodeBuilder)
```

Comment:
Cautious about this being a good approach long term, though I think to get the required amount of modularity could be really hard so happy for now.